src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs

```diff
@@ -361,6 +364,13 @@ internal static unsafe void FetchData<T>(IntPtr data, Span<T> result)
             dataSpan.CopyTo(result);
         }
 
+        internal static unsafe void FetchStringData<T>(TFTensor tensor, Span<T> result)
+        {
+            var buffer = TFTensor.DecodeStringTensor(tensor);
+            for (int i = 0; i < buffer.Length; i++)
+                result[i] = (T)(object)Encoding.UTF8.GetString(buffer[i]).AsMemory();
```

Comment:
>(T)(object) [](start = 28, length = 11)

is this double casting necessary?

Reply:
Yes, generic type needs casting this way.

---
In reply to: [257012507](https://github.com/dotnet/machinelearning/pull/2546#discussion_r257012507) [](ancestors = 257012507)